Mirror standard output plot class

braden6521 opened the pull request:
- Added a StandardPlotOuptut class to the CSP directory. This produces standard formatted plots. 
- Added example and unit tests for StandardPlotOutput.
- Added processing steps to curvature plots when viewing orthorectified images of a mirror. This helps make otherwise unviewablely noisy images understandable.

e10harvey commented on opencsp/common/lib/csp/StandardPlotOutput.py:
```diff
@@ -0,0 +1,556 @@
+            lt.info('Do not have both measured and reference optic; skipping slope deviation plots.')
+
+    def _plot_ray_trace_image_measured_optic(self):
+        """Plot ray trace image for measured optic"""
```
no need for doc-strings here. if you don't want this to potentially be pulled into the online documentation, I would convert to hash-tag comment.

e10harvey commented on opencsp/common/lib/csp/StandardPlotOutput.py:
```diff
@@ -0,0 +1,556 @@
+            lt.info('No reference optic; skipping reference optic ray trace.')
+
+    def _ray_trace_scene(self, obj: RayTraceable) -> rt.RayTrace:
+        """Performs a raytrace of a simple scene with a source and an optic.
```
Same as above. I'm fine with leaving these as doc-strings, but don't want to give the impression that this is a user-facing function.

e10harvey reviewed (approved):
In general everything looks great. I would again advocate for random values to be used in the tests, rather than hard coded numbers like 0, 7, and 100. The only issue I see is with using doc-strings rather than hash-tag comments for protected class members. If this is no longer a standard for us, we should clarify that in https://github.com/sandialabs/OpenCSP/blob/develop/doc/source/contributing.rst#coding-standards.

braden6521 commented:
@e10harvey. Thanks for your review. I've made the following updates, could you please re-review?

- I updated the docstrings to be hashtags for private members. 
- On that note, I also added more documentation to the StandardPlotOutput class itself (there were a few places where I thought it was lacking).
- I also added some more checks to the unit test. 
- I'm not sure how to add randomness into the unit test. Since we are comparing generated images to saved images, the inputs have to be the same every time. I can see cases where this could be employed in the future though on other tests.

e10harvey commented on opencsp/common/lib/csp/StandardPlotOutput.py:
```diff
@@ -0,0 +1,632 @@
+            self.fig_control, name='Ensquared Energy', number_in_name=self.options_file_output.number_in_name
+        )
+
+        # Dray reference if available
```
```suggestion
        # Draw reference if available
```

e10harvey reviewed (dismissed):
Thank you, @braden6521 !

braden6521 commented:
Thanks @e10harvey!
